packages/twenty-front/src/modules/settings/data-model/components/SettingsObjectFieldSelectForm.tsx

```diff
@@ -72,6 +74,45 @@ export const SettingsObjectFieldSelectForm = ({
     onChange(nextOptions);
   };
 
+  const handleUniqueDefaultValueChange = (
+    index: number,
+    option: SettingsObjectFieldSelectFormOption,
+    nextOption: SettingsObjectFieldSelectFormOption,
+  ) => {
+    const hasDefaultOptionChanged = option.isDefault !== nextOption.isDefault;
+    const nextOptions = hasDefaultOptionChanged
+      ? values.map((value) => ({
+          ...value,
+          isDefault: false,
+        }))
+      : [...values];
+
+    nextOptions.splice(index, 1, nextOption);
+
+    onChange(nextOptions);
+  };
+
+  const handleDefaultValueChange = (
+    index: number,
+    nextOption: SettingsObjectFieldSelectFormOption,
+  ) => {
+    const nextOptions = [...values];
+    nextOptions.splice(index, 1, nextOption);
+    onChange(nextOptions);
+  };
+
+  const findNewLabel = () => {
+    let optionIndex = values.length + 1;
+    while (optionIndex < 100) {
```

Comment:
this looks arbitrary to me but I don't have something really better

Reply:
I agree, but it is still better than what we had before.